internal/cli/atlas/customdbroles/update.go

```diff
@@ -63,30 +63,41 @@ func (opts *UpdateOpts) Run() error {
 	return opts.Print(out)
 }
 
-func (opts *UpdateOpts) newCustomDBRole(role *atlas.CustomDBRole) *atlas.CustomDBRole {
+func (opts *UpdateOpts) newCustomDBRole(existingRole *atlas.CustomDBRole) *atlas.CustomDBRole {
 	out := &atlas.CustomDBRole{
-		Actions:        []atlas.Action{},
-		InheritedRoles: []atlas.InheritedRole{},
-	}
-
-	if len(opts.action) > 0 {
-		out.Actions = convert.BuildAtlasActions(opts.action)
-	}
-
-	if len(opts.inheritedRoles) > 0 {
-		out.InheritedRoles = convert.BuildAtlasInheritedRoles(opts.inheritedRoles)
+		InheritedRoles: convert.BuildAtlasInheritedRoles(opts.inheritedRoles),
 	}
+	actions := convert.BuildAtlasActions(opts.action)
 
 	if opts.append {
-		out.Actions = append(out.Actions, role.Actions...)
-		out.InheritedRoles = append(out.InheritedRoles, role.InheritedRoles...)
+		actions = appendActions(existingRole.Actions, actions)
+		out.InheritedRoles = append(out.InheritedRoles, existingRole.InheritedRoles...)
 	}
+	out.Actions = actions
 
 	return out
 }
 
+func appendActions(existingActions, newActions []atlas.Action) []atlas.Action {
+	actionMap := make(map[string]atlas.Action)
+	for _, action := range existingActions {
+		actionMap[action.Action] = action
+	}
+	for i, action := range newActions {
+		if a, ok := actionMap[action.Action]; ok {
+			action.Resources = append(action.Resources, a.Resources...)
+			newActions[i] = action
```

Comment:
[not bloking] I personally don't like adding/removing/updating an element while iterating over a list.